onnxruntime/core/session/inference_session.cc

```diff
@@ -1432,7 +1440,12 @@ common::Status InferenceSession::Initialize() {
     }
 #endif  // !defined(ORT_MINIMAL_BUILD)
 
+    // Resolve the memory pattern flag of the main graph session state
     session_state_->ResolveMemoryPatternFlag();
+
+    // Resolve all memory pattern flags of subgraph session states
+    ResolveSubgraphMemoryPatternFlags(*session_state_);
+
     is_inited_ = true;
```

Comment:
nit: can we just call it ResolveMemoryPatternFlags and make all calls to SessionState::ResolveMemoryPatternFlag in it?